html/semantics/forms/the-select-element/common-HTMLOptionsCollection-namedItem.html

```diff
@@ -37,26 +39,16 @@
 }, "if no item has a name or id value matching the parameter, return null and stop");
 
 test(function () {
-    var testarr = [];
-    for (var i = 0; i < selly.namedItem('id3').length; i++) {
-        testarr.push(selly.namedItem('id3')[i].text);
-    }
-    assert_array_equals(testarr, ['3','duplicate ID']);
-}, "return an HTMLOptionsCollection in correct order for repeated 'id' value");
+    assert_equals(selly.namedItem('id3')["value"], "3");
+}, "if multiple items have a name or *id* value matching the parameter, return the first object and stop");
 
 test(function () {
-    var testarr = [];
-    for (var i = 0; i < selly.namedItem('name4').length; i++) {
-        testarr.push(selly.namedItem('name4')[i].text);
-    }
-    assert_array_equals(testarr, ['4', 'duplicate name']);
-}, "return an HTMLOptionsCollection in correct order for repeated 'name' value");
+    assert_equals(selly.namedItem('name4')["value"], "4");
+}, "if multiple items have a *name* or id value matching the parameter, return the first object and stop");
 
 test(function () {
-    var testarr = [];
-    for (var i = 0; i < selly.namedItem('mixed1').length; i++) {
-        testarr.push(selly.namedItem('mixed1')[i].text);
-    }
-    assert_array_equals(testarr, ['mixed ID', 'mixed name']);
-}, "return an HTMLOptionsCollection in correct order for repeated mixed value");
+    assert_equals(selly.namedItem('mixed1')["value"], "mixed ID");
```

Comment:
I'm reading the spec differently. That paragraph concerns property names of the ```select``` element. It's not apparent to me that it concerns the behavior of ```namedItem```.

In the text you cite, "supported property names" is linked to:

> The IDL fragments in this specification must be interpreted as required for conforming IDL fragments, as described in the Web IDL specification. [WEBIDL]
> 
> The following terms are defined in the Web IDL specification:
> 
> * ...
> * Supported property names
> * ...

[WEBIDL] is linked to https://heycam.github.io/webidl/. Searching that document for "supported property names", we get to:

> An interface that defines a named property getter is said to **support named properties**.
> 
> If an interface supports named properties, then the interface definition must be accompanied by a description of the ordered set of names that can be used to index the object at any given time. These names are called the **supported property names**.

In that light, for the ```select``` element in the test:

```
 <select id="selly">
   <option id="id1" name="name1">1</option>
   <option id="id2" name="name2">2</option>
   <option id="id3" name="name3">3</option>
   <option id="id4" name="name4">4</option>
   <option name="nameonly">nameonly</option>
   <option id="id3">duplicate ID</option>
   <option name="name4">duplicate name</option>
   <option id="mixed1">mixed ID</option>
   <option name="mixed1">mixed name</option>
 </select>
```

Enumerating ```selly```, we'd have:

```
selly.id1.text === 1
selly.name1.text === 1
selly.id2.text === 2
selly.name2.text === 2
selly.id3.text === 3
selly.name3.text === 3
selly.id4.text === 4
selly.name4.text === 4
selly.nameonly.text === "nameonly"
selly.mixed1.text === "mixed ID"
```

That's the gist of what that paragraph means to me. Even if I didn't get the rules exactly quite right, it seems to me that this concerns property names, not the behavior of ```namedItem```, and so should be covered by a different test.

That said, the following test seems to belie what I just said: https://github.com/w3c/web-platform-tests/blob/master/html/semantics/forms/the-select-element/select-named-getter.html. It seems to try to access the ```option``` elements via property name and assert them to be ```undefined```. So either that test is wrong, or my interpretation of the paragraph you cite is wrong.

What do you think? Am I off base?